File: Advanced Programming Methods/toyLanguage/toyLanguage/model/CompStmt.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace toyLanguage
{
    class CompStmt : IStmt
    {
        private IStmt first;
        private IStmt snd;

        public CompStmt(IStmt f, IStmt s)
        {
            first = f;
            snd = s;
        }

        public IStmt First{
            get{return first;}
            set { first = value; }
        }

        public IStmt Snd
        {
            get { return snd; }
            set { snd = value; }
        }

        /*public override PrgState Execute(PrgState state)
         {
             return state;
         }*/

        public override String ToString()
        {
            return "(" + first.ToString() + " " + snd.ToString()+")";
        }

        public PrgState Execute(PrgState state)
        {
            IMyStack<IStmt> stk = state.ExeStack;
            stk.Push(snd);
            stk.Push(first);
            return state;
        }
    }
}

File: Advanced Programming Methods/toyLanguage - Lock/src/sample/model/CompStmt.java
package sample.model;

import sample.utils.MyIStack;

public class CompStmt implements IStmt {
    IStmt first, second;
    public CompStmt(IStmt first, IStmt second){
        this.first=first;
        this.second=second;
    }
    @Override
    public PrgState execute(PrgState state){
        MyIStack stack=state.getExeStack();
        stack.push(second);
        stack.push(first);
        return null;
    }

    @Override
    public String toString(){
        return "("+first+" ; "+second+")";
    }
}
